### ISSUE TITLE
[Feature] Extend MLflow Integration to Walk-Forward, AutoTrader, and Online Learning

### ISSUE THREAD

**Issue by osama1998H** [osama1998H/Leap#94]
## Background

This issue captures the features from PR #60 (`claude/mlflow-integration-review-017nNFDSuvCHaacmDPT7UgFE`) which cannot be merged due to significant structural changes to the codebase since the PR was created:

1. **CLI Package Modularization (ADR-0008)**: `main.py` was refactored into the `cli/` package
2. **3 files with merge conflicts** that require substantial rework
3. **Integration points have moved** to different locations

The PR added ~952 lines across 6 files. This issue documents the features for clean reimplementation using current architecture patterns.

### Current State
- MLflow is integrated in **training** (`training/trainer.py`) and **backtest** (`cli/commands/backtest.py`)
- MLflow infrastructure exists in `utils/mlflow_tracker.py` (902 lines)
- Missing: Walk-forward, Monte Carlo, AutoTrader, and Online Learning logging

---

### A. Walk-Forward Optimization MLflow Logging

**Files to modify:**
- `utils/mlflow_tracker.py` - Add new methods
- `evaluation/backtester.py` - Accept optional `mlflow_tracker` parameter in `WalkForwardOptimizer`
- `cli/commands/walkforward.py` - Add MLflow context

**New methods in MLflowTracker:**
```python
def log_walkforward_params(self, train_window: int, test_window: int, step_size: int, n_splits: int) -> None:
    """Log walk-forward optimization parameters."""

def log_walkforward_fold(self, fold_idx: int, metrics: Dict[str, float], 
                          train_start: str, train_end: str, 
                          test_start: str, test_end: str) -> None:
    """Log metrics for a single walk-forward fold."""

def log_walkforward_summary(self, results: Dict[str, Any]) -> None:
    """Log aggregated walk-forward optimization results."""
```

**Metrics to log per fold:**
- total_return, sharpe_ratio, sortino_ratio, max_drawdown
- win_rate, profit_factor, total_trades, calmar_ratio, volatility

**Summary metrics:**
- n_folds, profitable_folds, profitable_ratio
- Aggregated metrics (mean, std, min, max) for each metric

---

### B. Monte Carlo Simulation MLflow Logging

**Files to modify:**
- `utils/mlflow_tracker.py` - Add new methods
- `evaluation/backtester.py` - Accept optional `mlflow_tracker` parameter in `MonteCarloSimulator`

**New methods in MLflowTracker:**
```python
def log_monte_carlo_params(self, n_simulations: int, confidence_level: float) -> None:
    """Log Monte Carlo simulation parameters."""

def log_monte_carlo_results(self, results: Dict[str, Any]) -> None:
    """Log Monte Carlo simulation results."""
```

**Metrics to log:**
- final_equity (mean, std, min, max, percentile_5, percentile_95)
- max_drawdown (mean, std, worst_case)
- probability_of_profit, probability_of_ruin

---

### C. AutoTrader MLflow Logging

**Files to modify:**
- `utils/mlflow_tracker.py` - Add new methods
- `core/auto_trader.py` - Accept optional `mlflow_tracker` parameter
- `cli/commands/autotrade.py` - Add MLflow context management

**New methods in MLflowTracker:**
```python
def log_autotrader_params(self, config: Dict[str, Any]) -> None:
    """Log auto-trader configuration parameters."""

def log_autotrader_session(self, stats: Dict[str, Any]) -> None:
    """Log final session statistics."""

def log_autotrader_heartbeat(self, balance: float, equity: float, 
                               positions: int, step: int) -> None:
    """Log periodic heartbeat metrics."""
```

**Params to log:**
- paper_mode, symbols, risk_per_trade, max_positions
- default_sl_pips, default_tp_pips, min_confidence, prediction_threshold
- enable_online_learning

**Session stats to log:**
- total_trades, winning_trades, losing_trades, total_pnl
- win_rate, max_drawdown
- signals_generated, signals_executed, signals_rejected
- online_adaptations

---

### D. Online Learning MLflow Logging

**Files to modify:**
- `utils/mlflow_tracker.py` - Add new methods
- `training/online_learning.py` - Accept optional `mlflow_tracker` parameter in `OnlineLearningManager`

**New methods in MLflowTracker:**
```python
def log_online_learning_params(self, config: Dict[str, Any]) -> None:
    """Log online learning configuration."""

def log_online_adaptation(self, adaptation_result: Dict[str, Any], step: int) -> None:
    """Log an adaptation event."""

def log_online_learning_step(self, metrics: Dict[str, float], step: int) -> None:
    """Log step metrics (periodically, e.g., every 100 steps)."""

def log_online_learning_report(self, report: Dict[str, Any]) -> None:
    """Log final online learning report."""
```

**Config params to log:**
- error_threshold, drawdown_threshold, performance_window
- adaptation_frequency, min_samples_for_adaptation
- learning_rate_decay, max_adaptations_per_day, regime_detection_enabled

**Step metrics to log:**
- prediction_error, trading_reward, sharpe_ratio, win_rate, profit_factor

---

### E. Feature Engineering MLflow Logging (Lower Priority)

**Files to modify:**
- `utils/mlflow_tracker.py` - Add new methods
- `core/data_pipeline.py` or `core/feature_registry.py`

**New methods in MLflowTracker:**
```python
def log_feature_engineering_params(self, n_features: int, feature_names: List[str],
                                     multi_timeframe_enabled: bool) -> None:
    """Log feature engineering parameters."""

def log_feature_statistics(self, feature_stats: Dict[str, Dict[str, float]]) -> None:
    """Log feature statistics (mean, std, missing_rate for top N features)."""
```

---

## Implementation Pattern

Follow the existing pattern from `cli/commands/backtest.py`:

```python
tracker = system.mlflow_tracker
if tracker and tracker.is_enabled:
    run_name = f"<command>-{symbol}-{timeframe}-{timestamp}"
    with tracker.start_run(run_name=run_name, tags={"command": "<command>", "symbol": symbol}):
        tracker.log_params({...})
        tracker.log_<feature>_results(results)
        tracker.log_artifact(results_file)
```

---

## Files Summary

| File | Changes |
|------|---------|
| `utils/mlflow_tracker.py` | Add ~500-600 lines (new logging methods) |
| `cli/commands/walkforward.py` | Add MLflow context (~30-40 lines) |
| `cli/commands/autotrade.py` | Add MLflow context management (~50 lines) |
| `evaluation/backtester.py` | Add optional `mlflow_tracker` param to WalkForwardOptimizer and MonteCarloSimulator |
| `core/auto_trader.py` | Add optional `mlflow_tracker` param, log on start/stop/heartbeat |
| `training/online_learning.py` | Add optional `mlflow_tracker` param, log on step/adaptation |

---

## Acceptance Criteria

- [ ] Walk-forward command logs params, per-fold metrics, and summary to MLflow
- [ ] Monte Carlo results are logged when `--monte-carlo` flag is used in backtest
- [ ] AutoTrader logs session params, periodic heartbeats, and final stats
- [ ] Online learning logs adaptation events and performance metrics
- [ ] All logging is optional and conditional on `tracker.is_enabled`
- [ ] No errors when MLflow is disabled
- [ ] Follows existing patterns in codebase
- [ ] Tests pass

---

## References

- Original PR: #60
- ADR-0008: CLI Package Modularization
- Pattern reference: `cli/commands/backtest.py` lines 103-131
